To-do list/Controllers/LoginViewController.swift

```diff
@@ -23,29 +23,91 @@ class LoginViewController: UIViewController {
         if passwordTextField.text?.isEmpty == true {
             return "Please enter your password"
         }
+        else if passwordTextField.text!.count < 8 {
+            return "Password must have at least 8 characters"
+        }
         return nil
-    }
+    }    
     
-    @IBAction func signInButton(_ sender: UIButton) {        
+    private func validateCredentials() -> Bool {
         let loginMessage = validateEmail()
         let passMessage = validatePass()
         
-        if loginMessage == nil && passMessage == nil {
-            let taskListVC = TaskListViewController.loadFromStoryboard(type: TaskListViewController.self)
-            self.navigationController?.pushViewController(taskListVC, animated: true)
-        } else {
+        if loginMessage != nil || passMessage != nil {
             var message = String()
             if let loginMessage = loginMessage {
                 message = loginMessage
             } else if let passMessage = passMessage {
                 message = passMessage
             }
             Alert.showBasic(title: message, vc: self)
+            return false
+        } else {
+            return true
+        }
+    }
+    
+    @IBAction func signInButton(_ sender: UIButton) {
+        if validateCredentials() {
+            let email = loginTextField.text ?? ""
+            let password = passwordTextField.text ?? ""
+            let query = "auth"
+            
+            APICaller.shared.executePostRequest(with: query,
```

Comment:
1. Rename to APIHelper
2. You don't need singleton pattern.

Reply:
done